core/dbt/parser/partial.py

```diff
@@ -968,13 +968,17 @@ def delete_schema_semantic_model(self, schema_file, semantic_model_dict):
             elif unique_id in self.saved_manifest.disabled:
                 self.delete_disabled(unique_id, schema_file.file_id)
 
-        metrics = schema_file.generated_metrics.copy()
```

Comment:
So here we were deleting all generated metrics instead of just the one related to a semantic model

Reply:
Correct.